src/mem/metaslab.h

```diff
@@ -91,6 +91,32 @@ namespace snmalloc
       return reinterpret_cast<SlabLink*>(pointer_offset(slab, link));
     }
 
+    /// Value used to check for corruptions in a block
+    static constexpr size_t POISON = 
+      static_cast<size_t>(bits::is64() ? 0xDEADBEEFDEAD0000 : 0xDEAD0000);
```

Comment:
This won’t detect corruption in the low 16 bits. On low endian that’s the first two bytes of the object, which makes overwriting just those likely enough.
How about some sort of simple checksum? eg. repeat the two bytes in the remaining 6, and xor the poison.

Reply:
I think it would be cleaner to have an `add_poison` and a `check_and_remove_poison` method, then we could experiment with different poison mechanisms (ARM Pointer Authentication would be fun!)

Reply:
I am not sure on an API we want as this is currently restricted to 16-bit slab relative pointers.  Other mechanisms, would perhaps want different assumptions.  The code here factors out the functionality we need to provide in the free blocks. I think this is sufficient to make future experiments without much hassle.